src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,87 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.models.Booking;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.logging.Logger;
+
+@Service
+public class BookingService implements IBookingService {
+    private static final Logger LOGGER = Logger.getLogger(BookingService.class.getName());
+    private static final String COLLECTION_PATH = "bookings";
+
+    private Firestore db;
+
+    public void setDb(Firestore db) {
+        this.db = db;
+    }
+
+    /**
+     * Retrieves a Booking entity by its id.
+     */
+    @Override
+    public Booking getBookingByID(final UUID bookingId) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (bookingId == null) throw new IllegalArgumentException();
+        DocumentReference documentReference =  db.collection(COLLECTION_PATH)
+                .document(bookingId.toString());
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot document = future.get();
+        Booking booking;
+        if(document.exists()){
+            booking = document.toObject(Booking.class);
+            return booking;
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves a Booking entity by its user's id.
+     */
+    @Override
+    public Set<Booking> getBookings(final UUID userId) throws ExecutionException, InterruptedException, IllegalArgumentException  {
+        if (userId == null) throw new IllegalArgumentException();
+        Set<Booking> userBookings;
+        userBookings = db.collection(COLLECTION_PATH)
+                .whereEqualTo("accountId", userId).get().get()
+                .getDocuments()
+                .stream().map(d -> d.toObject(Booking.class))
+                .collect(Collectors.toSet());
+        return userBookings;
+    }
+
+    /**
+     * Retrieves all entities of type Booking.
+     */
+    @Override
+    public Set<Booking> getAllBookings() throws ExecutionException, InterruptedException {

```

Comment:
The reason why we don't want to Set<Booking>, because the set returns the booking history with unordered manner. Usually the user wants to list their history with chronological order. For this problem we usually use LinkedHashSet<>.
https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashSet.html